Supersonic crash during scrobble

How to reproduce: 
1. Enable Shuffle
2. Navigate to 'All Tracks' and start playback on any track
3. Via the play queue, select and play a track later in the queue (this needs to have an index which is larger than the album size in the next step)
4. Navigate to any album which has a track size less than the index of the currently playing track
5. Play any track in the album by double-clicking it
6. Crash with following stack trace:

```
panic: runtime error: index out of range [25] with length 11

goroutine 23 [running]:
github.com/dweymouth/supersonic/backend.(*playbackEngine).getPlayQueueItemAt(...)
        D:/projects/supersonic/backend/playbackengine.go:280
github.com/dweymouth/supersonic/backend.(*playbackEngine).checkScrobble(0x0?)
        D:/projects/supersonic/backend/playbackengine.go:1051 +0x2e5
github.com/dweymouth/supersonic/backend.(*playbackEngine).playTrackAt(0xc0003be008, 0x0, 0x0)
        D:/projects/supersonic/backend/playbackengine.go:348 +0xba
github.com/dweymouth/supersonic/backend.(*playbackEngine).PlayTrackAt(...)
        D:/projects/supersonic/backend/playbackengine.go:310
github.com/dweymouth/supersonic/backend.(*playbackEngine).LoadItemsAndPlayAtIdx(0xc0003be008, {0xc003833970, 0xb, 0xb}, 0x0, 0x6)
        D:/projects/supersonic/backend/playbackengine.go:592 +0x265
github.com/dweymouth/supersonic/backend.(*PlaybackManager).runCmdQueue(0xc000d84000, {0x7ff7f8b9c8e8, 0xc000348000})
        D:/projects/supersonic/backend/playbackmanager.go:906 +0xd29
created by github.com/dweymouth/supersonic/backend.NewPlaybackManager in goroutine 1
        D:/projects/supersonic/backend/playbackmanager.go:83 +0x459
```

This happens because in shuffle-mode, the play queue is updated before `PlayTrackAt(...)` is called, which means that `checkScrobble()` tries to access the `playQueue` at the old `nowPlayingIdx`, which is larger than the current `playQueue`.

--- a/backend/playbackengine.go
+++ b/backend/playbackengine.go
@@ -262,6 +262,7 @@ func (p *playbackEngine) getPlayQueueLength() int {
 }
 
 func (p *playbackEngine) clearPlayQueue() {
+	p.checkScrobble()
 	p.player.Stop(false)
 	p.nowPlayingIdx = -1
 	p.playQueue = nil

--- a/backend/playbackengine.go
+++ b/backend/playbackengine.go
@@ -576,12 +576,13 @@ func (p *playbackEngine) LoadItems(items []mediaprovider.MediaItem, insertQueueM
 	return p.doLoaditems(newItems, insertQueueMode, shuffle)
 }
 
-// Load items into the play queue.
-// If replacing the current queue (!appendToQueue), playback will be stopped.
+// Load items into the play queue and play the track at idx.
+// Replaces the playQueue, shuffledPlayQueue, or both
 func (p *playbackEngine) LoadItemsAndPlayAtIdx(items []mediaprovider.MediaItem, shuffle bool, idx int) error {
 	newItems := deepCopyMediaItemSlice(items)
 
 	if p.shuffle || shuffle {
+		p.clearPlayQueue()
 		rand.Shuffle(len(newItems), func(i, j int) {
 			newItems[i], newItems[j] = newItems[j], newItems[i]
 		})

